core/server/api_container/server/startosis_engine/kurtosis_types/service_config/ready_conditions.go

```diff
@@ -0,0 +1,231 @@
+	assertion, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[starlark.String](readyConditions.KurtosisValueTypeDefault, AssertionAttr)
+	if interpretationErr != nil {
+		return "", interpretationErr
+	}
+	if !found {
+		return "", startosis_errors.NewInterpretationError("Required attribute '%s' could not be found on type '%s'",
+			AssertionAttr, ReadyConditionsTypeName)
+	}
+	assertionStr := assertion.GoString()
+
+	return assertionStr, nil
+}
+
+func (readyConditions *ReadyConditions) GetTarget() (starlark.Comparable, *startosis_errors.InterpretationError) {
+	target, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[starlark.Comparable](readyConditions.KurtosisValueTypeDefault, TargetAttr)
+	if interpretationErr != nil {
+		return nil, interpretationErr
+	}
+	if !found {
+		return nil, startosis_errors.NewInterpretationError("Required attribute '%s' could not be found on type '%s'",
+			TargetAttr, ReadyConditionsTypeName)
+	}
+
+	return target, nil
+}
+
+func (readyConditions *ReadyConditions) GetInterval() (time.Duration, *startosis_errors.InterpretationError) {
+	interval := defaultInterval
+
+	intervalStr, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[starlark.String](readyConditions.KurtosisValueTypeDefault, IntervalAttr)
+	if interpretationErr != nil {
+		return interval, interpretationErr
+	}
+	if found {
+		parsedInterval, parseErr := time.ParseDuration(intervalStr.GoString())
+		if parseErr != nil {
+			return interval, startosis_errors.WrapWithInterpretationError(parseErr, "An error occurred when parsing interval '%v'", intervalStr.GoString())
+		}
+		interval = parsedInterval
+	}
+
+	return interval, nil
+}
+
+func (readyConditions *ReadyConditions) GetTimeout() (time.Duration, *startosis_errors.InterpretationError) {
+	timeout := defaultTimeout
+
+	timeoutStr, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[starlark.String](readyConditions.KurtosisValueTypeDefault, TimeoutAttr)
+	if interpretationErr != nil {
+		return timeout, interpretationErr
+	}
+	if found {
+		parsedTimeout, parseErr := time.ParseDuration(timeoutStr.GoString())
+		if parseErr != nil {
+			return timeout, startosis_errors.WrapWithInterpretationError(parseErr, "An error occurred when parsing timeout '%v'", timeoutStr.GoString())
+		}
+		timeout = parsedTimeout
+	}
+
+	return timeout, nil
+}
+
+func validateRecipe(value starlark.Value) *startosis_errors.InterpretationError {
+	_, ok := value.(*recipe.HttpRequestRecipe)
+	if !ok {
+		//TODO we should rework the recipe types to inherit a single common type, this will avoid the double parsing here.
+		_, ok := value.(*recipe.ExecRecipe)
+		if !ok {
+			return startosis_errors.NewInterpretationError("The '%s' attribute is not a Recipe (was '%s').", RecipeAttr, reflect.TypeOf(value))
+		}
+	}
+	return nil
+}
+
+func validateDuration(value starlark.Value, attributeName string) *startosis_errors.InterpretationError {
```

Comment:
We have a file where we implement generic validator, maybe you could add this one there, seems fairly generic. At least we will need that when we migrate recipe object types